spec/generator_spec.rb

```diff
@@ -55,35 +42,80 @@
+  end
+
+end
+
+shared_examples 'ECv2' do
+  context 'signed_key' do
+    let(:message) { instance.build_signed_key }
+    let(:result) { Aliquot::Validator::SignedKeyContract.schema.call(message) }
+
+    it 'validates' do
+      expect(result.success?).to be(true), result.errors.to_s
     end
 
-    it 'excludes intermediateSigningKey' do
-      expect(instance.token['intermediateSigningKey']).to be nil
+    it 'has the correct protocolVersion' do
+      expect(instance.token['protocolVersion']).to eq('ECv2')
     end
   end
+end
 
-  context :ECv2 do
-    let(:instance)     { AliquotPay.new(:ECv2) }
-    include_examples 'generation tests'
+describe AliquotPay do
+  context :ECv1 do
+    context 'non-tokenized' do
+      let(:instance) { AliquotPay.new(protocol_version: :ECv1, type: :browser) }
+      let(:details) { instance.build_payment_method_details }
+      include_examples 'generation tests'
+      it 'validates when type is browser' do
+        result = Aliquot::Validator::ECv1_PaymentMethodDetailsContract.schema.call(details)
+        expect(result.success?).to be(true), result.errors.to_s
+      end
+    end
+
+    context 'tokenized' do
+      let(:instance) { AliquotPay.new(protocol_version: :ECv1, type: :app) }
+      let(:details) { instance.build_payment_method_details }
+      include_examples 'generation tests'
+      it 'validates when type is app' do
+        details.merge!('threedsCryptogram' => details.delete('3dsCryptogram'))
+        result = Aliquot::Validator::ECv1_TokenizedPaymentMethodDetailsContract.schema.call(details)
+        expect(result.success?).to be(true), result.errors.to_s
+      end
+    end
+  end
 
-    context 'signed_key' do
-      let(:message)  { instance.build_signed_key }
-      let(:result)   { Aliquot::Validator::SignedKeySchema.call(message) }
 
-      it 'validates' do
+  context :ECv2 do
+    context 'non-tokenized' do
+      let(:instance) { AliquotPay.new(protocol_version: :ECv2, type: :browser) }
+      let(:details) { instance.build_payment_method_details }
+      let(:result)  { Aliquot::Validator::ECv2_PaymentMethodDetailsContract.schema.call(details) }
+      include_examples 'generation tests'
+      include_examples 'ECv2'
+      it 'validates when browser' do
         expect(result.success?).to be(true), result.errors.to_s
       end
     end
 
-    it 'has the correct protocolVersion' do
-      expect(instance.token['protocolVersion']).to eq('ECv2')
+    context 'tokenized' do
+      let(:instance) { AliquotPay.new(protocol_version: :ECv2, type: :app) }
+      let(:details) { instance.build_payment_method_details }
+      let(:result)  { Aliquot::Validator::ECv2_TokenizedPaymentMethodDetailsContract.schema.call(details) }
+      include_examples 'generation tests'
+      include_examples 'ECv2'
+      it 'validates when CRYPTOGRAM_3DS' do
+        expect(result.success?).to be(true), result.errors.to_s
+      end
     end
+
+
```

Comment:
I would remove these two empty lines. It seems I cannot make a real suggestion for it. It's hereby suggested. But remember all of my inputs are merely suggestions.